src/garage/tf/algos/tnpg.py

```diff
@@ -65,21 +65,14 @@ def __init__(self,
                  pg_loss='surrogate',
                  lr_clip_range=0.01,
                  max_kl_step=0.01,
-                 optimizer=None,
-                 optimizer_args=None,
+                 optimizer=ConjugateGradientOptimizer,
+                 optimizer_args=dict(max_backtracks=1),
                  policy_ent_coeff=0.0,
                  use_softplus_entropy=False,
                  use_neg_logli_entropy=False,
                  stop_entropy_gradient=False,
                  entropy_method='no_entropy',
                  name='TNPG'):
-        if optimizer is None:
-            optimizer = ConjugateGradientOptimizer
-            default_args = dict(max_backtracks=1)
-            if optimizer_args is None:
-                optimizer_args = default_args
-            else:
-                optimizer_args = dict(default_args, **optimizer_args)
```

Comment:
According to the logic here we want to keep `max_backtracks=1 ` for the `optimizer_args ` even if the user passes their own args. I'm not sure if it's necessary to be this value for this particular algorithm. (I traced back to `ConjugateGradientOptimizer` where the default `max_backtracks` is 15.)

Reply:
It is probably necessary. I think setting `max_backtracks` too large will probably affect the training performance of TNPG since it doesn't use the backtrack line search to make sure that the trust region is not broken.

Reply:
The danger here is that if the user passes some args other than `max_backtracks` it will override the default value. But the previous one (using `default_args`) also has a flaw: `max_backtracks` can not be customised otherwise it raises error for duplicate attribute values.

Reply:
No. The previous one won't raise duplicate attribute because python will overwrite the value of an existing key.

Reply:
I don't think the previous one is good because the values in `optimizer_args` is likely to overwrite the values in `default_args`.

Reply:
I prefer the current because I don't think we should constrain the `max_backtracks=1` or `max_epochs=1` in algorithms. It is unlikely that the algorithm won't be working when `max_epochs=2`. 
If we need fix this, I think set sane defaults in optimizer is better? For instance, the default `max_epochs` in `FirstOrderOptimizer` is 100, that is not a sane default. Even for a small and simple environment, running each iteration in 100 epochs is going to take a lot of time.

Reply:
Sorry, on second thought, I don't really have a preference. I think each has its drawbacks. If it is ok with you, I would like to revert this change (both `optimizer` and `optimizer_args`). I think it is not very important for this PR.

Reply:
Sure. Ok with me.